scalding-core/src/main/scala/com/twitter/scalding/TypedPipeChecker.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.scalding
+
+/**
+ * This class is used to assist with testing a TypedPipe
+ */
+object TypedPipeChecker {
+
+  /*
+   * Takes a List and a transform function.
+   * The resulting TypedPipe form the transform will be run through asserts
+   */
+  def checkOutputTransform[T, U, R](input: List[T])(transform: TypedPipe[T] => TypedPipe[U])(assertions: List[U] => R): R =
+    assertions(checkOutputInline(transform(TypedPipe.from(input))))
+
+  /*
+   * Execute a TypedPipe in memory, convert the resulting Iterator to
+   * a list and run it through a function that makes arbitrary
+   * assertions on it.
+   */
+  def checkOutput[T, R](output: TypedPipe[T])(assertions: List[T] => R): R =
+    assertions(checkOutputInline(output))
+
+  /**
+   * Execute a TypedPipe in memory and return the result as a List
+   */
+  def checkOutputInline[T](output: TypedPipe[T]): List[T] =
```

Comment:
I like this name for this function.  I kind of like idea of keeping the names for the others though since it really is just a nice to have utility function for running some asserts on a typed pipe and the name indicates that it probably should only be used for unit testing.

If you think it could be used outside of tests though we can try to think of some better names.